client/my-sites/checkout/composite-checkout/components/prepurchase-notices/index.jsx

```diff
@@ -72,39 +69,33 @@ const PrePurchaseNotices = () => {
 		return products.filter( ( p ) => ! p.expired );
 	} );
 
-	/**
-	 * The active product on the current site that overlaps/conflicts with the plan currently in the cart.
-	 */
-	const siteProductThatOverlapsCartPlan = useSelector( ( state ) => {
+	const siteProductThatOverlapsCartPlan = useMemo( () => {
 		const planSlugInCart = cartItemSlugs.find( isJetpackPlanSlug );
-
-		if ( ! planSlugInCart ) {
+		if ( ! planSlugInCart || ! currentSiteProducts ) {
 			return null;
 		}
 
-		if (
-			planHasFeature( planSlugInCart, WPCOM_FEATURES_BACKUPS ) &&
-			siteHasFeature( state, siteId, WPCOM_FEATURES_BACKUPS )
-		) {
-			return currentSiteProducts.find( isJetpackBackup );
-		}
+		const getMatchingProducts = ( siteProducts, planSlug ) => {
+			// Get all features and products for the plan in the cart
+			const planFeatures = getAllFeaturesForPlan( planSlug );
 
-		if (
-			planHasFeature( planSlugInCart, WPCOM_FEATURES_ANTISPAM ) &&
-			siteHasFeature( state, siteId, WPCOM_FEATURES_ANTISPAM )
-		) {
-			return currentSiteProducts.find( isJetpackAntiSpam );
-		}
+			// Filter the site products to only include those in the plan items or are inferior features to the plan feature
+			const matchingProducts = siteProducts.filter(
+				( product ) =>
+					planFeatures.includes( product.productSlug ) ||
+					planHasSuperiorFeature( planSlug, product.productSlug )
+			);
 
-		if (
-			planHasFeature( planSlugInCart, WPCOM_FEATURES_SCAN ) &&
-			siteHasFeature( state, siteId, WPCOM_FEATURES_SCAN )
-		) {
-			return currentSiteProducts.find( isJetpackScan );
+			return matchingProducts;
+		};
+
+		const matchingProducts = getMatchingProducts( currentSiteProducts, planSlugInCart );
+		if ( matchingProducts.length ) {
+			return matchingProducts[ 0 ];
 		}
 
 		return null;
```

Comment:
Non-blocking: there's a shorter way to write this 🙂

```suggestion
		return matchingProducts?.[ 0 ];
```

Reply:
👍